### ISSUE TITLE
minikube kubect: permission denied

### ISSUE THREAD

**Issue by bartoszj** [kubernetes/minikube#6583]
After upgrading to minikube v.1.7.2 I'm not longer able to use `minikube kubectl`.

**The exact command to reproduce the issue**:
```
rm -rf ~/.minikube
minikube start
minikube kubectl
```

**The full output of the command that failed**:<details>
```
Error running /Users/<USER>/.minikube/cache/v1.17.2/kubectl: fork/exec /Users/<USER>/.minikube/cache/v1.17.2/kubectl: permission denied
```

</details>

**Contents of the `~/.minikube/cache/v1.17.2/kubectl`**:<details>
```
ls -l ~/.minikube/cache/v1.17.2
-rw-------  1 <USER>  staff   39342080 11 lut 10:27 kubeadm
-rw-------  1 <USER>  staff   43491328 11 lut 10:27 kubectl
-rw-------  1 <USER>  staff  111568408 11 lut 10:27 kubelet
```

```
file ~/.minikube/cache/v1.17.2/kubectl
/Users/<USER>/.minikube/cache/v1.17.2/kubectl: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), statically linked, Go BuildID=itoADCJwxOm7z3ymMN2Z/yoeLs_MYMz8Q5D0z_bWX/K9sqErNfMu6XJcqYVzr4/nHZFbmivkI7eipSyzj5i, stripped
```

</details>

**The operating system version**: macOS Catalina 10.15.3 (19D76)

It looks like minikube downloaded executables for Linux operating system, where it should download executable to macOS. Currently the workaround is to delete files from `~/.minikube/cache/v1.17.2` folder and execute `minikube kubectl` which will download correct executables for macOS.

**Comment by afbjorklund** [comment]
Same as #6559 - but in reverse

**Comment by afbjorklund** [comment]
Fixed in v1.7.3
